next/src/modules/mention/components/MentionField.js

```diff
@@ -0,0 +1,43 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { Mention } from 'react-mentions'
+
+import MentionInput from './MentionInput'
+import UsersMentionContainer from '../containers/UsersMentionContainer'
+import { MENTION_PATTERN } from '../lib/const'
+
+const MentionField = ({ input, ...props }) => (
+  <UsersMentionContainer.Fetch>
+    {fetchUsers => {
+      const transform = ({ data }) =>
+        data.users.entities
+          .map(user => user ? ({ display: user.name, id: user.uid }) : false)
+          .filter(Boolean)
+
+      const data = (query, cb) =>
```

Comment:
Is there a special reason to fire the query programatically?

Reply:
Sim por causa da api do react-mentions. Usasse a prop `data` pra fazer dar as suggestions de forma assíncrona, retornando uma Promise.